kong/plugins/bot-detection/handler.lua

```diff
@@ -49,7 +49,7 @@ function BotDetectionHandler:access(conf)
 
     if conf.whitelist then
       for _, rule in ipairs(conf.whitelist) do
-        if re_match(user_agent, rule) then
+        if re_find(user_agent, rule) then
```

Comment:
The scope of this commit being focused on performance for regex evaluation, it'd be good to go the extra mile and introduce the PCRE JIT flags here as well, in the same `perf` commit: `"jo"`